app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,26 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    if user && (user.token_set_time  >= 2.hours.ago)
```

Comment:
I would recommend abstracting the length of time into a setting. This could be in a constant in your class, an environment variable, or in another module via initializer if you were so inclined.